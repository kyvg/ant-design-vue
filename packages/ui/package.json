{
  "name": "@ant-design-vue/ui",
  "version": "1.0.0-alpha.1",
  "title": "Ant Design Vue",
  "description": "An enterprise-class UI design language and Vue-based implementation",
  "keywords": [
    "vue",
    "vue3",
    "ant",
    "design",
    "antd",
    "vueComponent",
    "component",
    "components",
    "ui",
    "framework",
    "frontend"
  ],
  "scripts": {
    "build": "vite build",
    "build:dev": "vite build --mode development",
    "dev": "vite build --watch --mode development",
    "lint": "eslint . --fix",
    "tsc": "tsc --noEmit",
    "tsg": "tsc --declaration --declarationMap --emitDeclarationOnly --noEmit false --outDir dist/types"
  },
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist/types/index.d.ts",
      "import": "./dist/lib.mjs",
      "require": "./dist/lib.cjs",
      "default": "./dist/lib.mjs"
    },
    "./style.css": {
      "import": "./dist/lib.css",
      "require": "./dist/lib.css",
      "default": "./dist/lib.css"
    },
    "./tailwind.css": {
      "import": "./dist/tailwind.css",
      "require": "./dist/tailwind.css",
      "default": "./dist/tailwind.css"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ant-design-vue/core.git"
  },
  "bugs": {
    "url": "https://github.com/ant-design-vue/table/issues"
  },
  "license": "MIT",
  "main": "./dist/lib.cjs",
  "module": "./dist/lib.mjs",
  "types": "./dist/types/index.d.ts",
  "files": [
    "dist",
    "src/style",
    "typings"
  ],
  "sideEffects": [
    "dist/*",
    "*.less"
  ],
  "peerDependencies": {
    "vue": "^3.2.34"
  },
  "dependencies": {
    "@ant-design/colors": "^7.0.0",
    "@ant-design/icons-vue": "^6.0.1",
    "@floating-ui/dom": "^1.6.13",
    "@floating-ui/vue": "^1.1.6",
    "lodash-es": "^4.17.21",
    "resize-observer-polyfill": "^1.5.1"
  },
  "devDependencies": {
    "@ant-design-vue/eslint-config": "*",
    "@ant-design-vue/prettier-config": "*",
    "@ant-design-vue/typescript-config": "*",
    "@ant-design-vue/vite-config": "*",
    "@tailwindcss/vite": "^4.1.3",
    "@vitejs/plugin-vue": "^5.1.3",
    "@vueuse/core": "^12.0.0",
    "@types/node": "^20.19.7",
    "prettier-plugin-tailwindcss": "^0.6.11",
    "tailwindcss": "^4.0.14",
    "typescript": "^5.8.2",
    "vite": "^5.3.5",
    "vite-plugin-dts": "^4.5.4",
    "vite-svg-loader": "^5.1.0",
    "vue": "^3.4.34",
    "vue-tsc": "^3.0.3"
  }
}
