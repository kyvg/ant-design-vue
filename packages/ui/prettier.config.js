// @ts-check

export { default } from "@ant-design-vue/prettier-config/tailwind";
