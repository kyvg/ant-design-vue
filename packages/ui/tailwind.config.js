/** @type {import('tailwindcss').Config} */
export default {
  content: ['./src/**/*.{vue,js,ts,jsx,tsx}', './src/**/*.css'],
  safelist: [
    // 强制包含所有主题颜色类
    {
      pattern:
        /^(text|bg|border)-(base|primary|secondary|accent|neutral|info|success|warning|error)(-\d+)?$/,
      variants: ['hover', 'focus', 'active', 'disabled'],
    },
    {
      pattern:
        /^(text|bg|border)-(base|primary|secondary|accent|neutral|info|success|warning|error)-content$/,
      variants: ['hover', 'focus', 'active', 'disabled'],
    },
  ],
  theme: {
    extend: {
      colors: {
        base: {
          100: 'var(--color-base-100)',
          200: 'var(--color-base-200)',
          300: 'var(--color-base-300)',
          content: 'var(--color-base-content)',
        },
        primary: {
          DEFAULT: 'var(--color-primary)',
          content: 'var(--color-primary-content)',
        },
        secondary: {
          DEFAULT: 'var(--color-secondary)',
          content: 'var(--color-secondary-content)',
        },
        accent: {
          DEFAULT: 'var(--color-accent)',
          content: 'var(--color-accent-content)',
        },
        neutral: {
          DEFAULT: 'var(--color-neutral)',
          content: 'var(--color-neutral-content)',
        },
        info: {
          DEFAULT: 'var(--color-info)',
          content: 'var(--color-info-content)',
        },
        success: {
          DEFAULT: 'var(--color-success)',
          content: 'var(--color-success-content)',
        },
        warning: {
          DEFAULT: 'var(--color-warning)',
          content: 'var(--color-warning-content)',
        },
        error: {
          DEFAULT: 'var(--color-error)',
          content: 'var(--color-error-content)',
        },
      },
    },
  },
}
