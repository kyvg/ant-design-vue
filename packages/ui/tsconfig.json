{
  "extends": "@ant-design-vue/typescript-config/tsconfig.vue.json",
  "include": ["src/**/*.ts", "src/**/*.vue", "src/**/*.d.ts"],
  "references": [{ "path": "./tsconfig.node.json" }],
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"],
      "~/*": ["./assets/*"]
    }
  }
}
