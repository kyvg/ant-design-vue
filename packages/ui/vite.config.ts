import { extendsConfig } from '@ant-design-vue/vite-config'
import vue from '@ant-design-vue/vite-config/vue'
import { resolve } from 'node:path'
import tailwindcss from '@tailwindcss/vite'
import { readdirSync, statSync } from 'node:fs'
import dts from 'vite-plugin-dts'
import pkg from './package.json'

// 获取所有组件目录
function getComponents() {
  const componentsDir = resolve(__dirname, 'src/components')
  const components: Record<string, string> = {}

  try {
    const entries = readdirSync(componentsDir)
    entries.forEach(entry => {
      const fullPath = resolve(componentsDir, entry)
      if (statSync(fullPath).isDirectory()) {
        components[`components/${entry}`] = resolve(fullPath, 'index.ts')
      }
    })
  } catch (error) {
    console.warn('Failed to read components directory:', error)
  }

  return components
}

const components = getComponents()

export default extendsConfig(vue(__dirname, true), {
  resolve: {
    alias: {
      '~': resolve(__dirname, './assets'),
    },
  },
  plugins: [tailwindcss()],
  test: {
    alias: {
      [pkg.name]: resolve(__dirname, './src'),
    },
    globals: true,
    environment: 'jsdom',
    testTimeout: 5000,
    coverage: {
      include: ['src/**/*.{ts,vue}'],
      provider: 'v8',
      allowExternal: true,
      reporter: ['text', 'json', 'html'],
    },
    setupFiles: [resolve(__dirname, './test/setup.ts')],
  },
  build: {
    cssCodeSplit: true,
    lib: {
      entry: {
        lib: resolve(__dirname, 'src/index.ts'),
        ...components,
      },
      formats: ['es', 'cjs'],
      fileName: (format, entryName) => `${entryName}.${format === 'es' ? 'mjs' : 'cjs'}`,
    },
    rollupOptions: {
      output: {
        manualChunks(id) {
          if (id.includes('tailwind.css')) {
            return 'tailwind'
          }
          if (id.includes('.css')) {
            return 'lib'
          }
        },
      },
    },
  },
})
