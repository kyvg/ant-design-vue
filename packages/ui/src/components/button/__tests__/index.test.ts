import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { Button } from '@ant-design-vue/ui'
import { mount } from '@vue/test-utils'

describe('Button', () => {
  it('should render correctly', () => {
    const wrapper = mount(Button)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
