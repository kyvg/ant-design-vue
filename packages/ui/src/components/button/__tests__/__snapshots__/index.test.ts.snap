// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render correctly 1`] = `
"<button class="ant-btn ant-btn-solid ant-btn-md">
  <!--v-if-->
  <!--v-if--><span></span>
</button>"
`;
